ex1.py

```diff
@@ -1,3 +1,13 @@
-if __name__ == '__main__':
-    print("5")
-    print("5")
\ No newline at end of file
+
+
+def avg():
+    counter = 0
+    sum = 0
+    num = input("enter num if you want to stop enter q ")
+    while num != "q" :
+        sum += float(num)
```

Comment:
what  happens if num isnt a number and the user entered a character instead?

please use try and except to catch the exception.
and print a message that explains to the user he entered bad input

Reply:
i need the add something that saying that character is a bad input

Reply:
yes but use exceptions.
take you code as is and try to enter a string instead of a number, see what happens.

spoiler: you will see that an exception is raised. add an try and except clause so when the exception is raised whats in the except clause will be excecuted.

you can also read about exceptions in general:

[https://realpython.com/python-exceptions/](url)